src/prediction_market/views/pred_market_view.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import viewsets
+from rest_framework.filters import OrderingFilter
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+
+from prediction_market.models import PredictionMarket
+from prediction_market.serializers.prediction_market_serializer import (
+    DynamicPredictionMarketSerializer,
+    PredictionMarketSerializer,
+)
+from prediction_market.utils import create_prediction_market
+from utils.throttles import THROTTLE_CLASSES
+
+
+class PredictionMarketViewSet(viewsets.ModelViewSet):
+    serializer_class = PredictionMarketSerializer
```

Comment:
If we store the values of votes / bets of the PredictionMarket inside the model itself, we'll need a way for users to not be able to edit that model.

Reply:
Set `permission_classes` to only allow Moderators, and overwrote any write/update methods to return `405: Not Implemented`

Reply:
If the permission_classes there is only moderators, will regular people be able to vote Yes or No?

Reply:
Yes, there's a separate view for votes with `IsAuthenticatedOrReadOnly`